evm/src/NttManager/NttManagerWithPerChainTransceivers.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity >=0.8.8 <0.9.0;
+
+import "./NttManagerNoRateLimiting.sol";
+
+/// @title  NttManagerWithPerChainTransceivers
+/// @author Wormhole Project Contributors.
+/// @notice The NttManagerWithPerChainTransceivers abstract contract is an implementation of
+///         NttManager that allows configuring different transceivers and thresholds
+///         for each chain. You can configure a different set of send and receive
+///         transceivers for each chain, and if you don't specifically enable any
+///         transceivers for a chain, then all transceivers will be used for it.
+///
+/// @dev    All of the developer notes from `NttManager` apply here.
+contract NttManagerWithPerChainTransceivers is NttManagerNoRateLimiting {
+    constructor(
+        address _token,
+        Mode _mode,
+        uint16 _chainId
+    ) NttManagerNoRateLimiting(_token, _mode, _chainId) {}
+
+    // ==================== Storage slots ==========================================================
+
+    bytes32 private constant SEND_TRANSCEIVER_BITMAP_SLOT =
+        bytes32(uint256(keccak256("nttpct.sendTransceiverBitmap")) - 1);
+
+    bytes32 private constant RECV_TRANSCEIVER_BITMAP_SLOT =
+        bytes32(uint256(keccak256("nttpct.recvTransceiverBitmap")) - 1);
+
+    // ==================== Override / implementation of transceiver stuff =========================
+
+    /// @inheritdoc IManagerBase
+    function enableSendTransceiverForChain(
+        address transceiver,
+        uint16 forChainId
+    ) external override(ManagerBase, IManagerBase) onlyOwner {
+        _enableTransceiverForChain(transceiver, forChainId, SEND_TRANSCEIVER_BITMAP_SLOT);
+        emit SendTransceiverEnabledForChain(transceiver, forChainId);
+    }
+
+    /// @inheritdoc IManagerBase
+    function enableRecvTransceiverForChain(
+        address transceiver,
+        uint16 forChainId
+    ) external override(ManagerBase, IManagerBase) onlyOwner {
+        _enableTransceiverForChain(transceiver, forChainId, RECV_TRANSCEIVER_BITMAP_SLOT);
+        emit RecvTransceiverEnabledForChain(transceiver, forChainId);
+    }
+
+    function _enableTransceiverForChain(
```

Comment:
I think we should also add the inverse `_disableTransceiverForChain` with associated entrypoints?

Reply:
Is this really necessary (worth the storage)? Is an integrator ever likely to disable something?

Reply:
I will add the disable methods.

Reply:
This comment was made obsolete by the switch to use get/set instead of enable/disable.